MyHeartCounts/Health Import/HistoricalHealthSamplesExportManager.swift

```diff
@@ -0,0 +1,170 @@
+    @Dependency(Account.self)
+    private var account: Account?
+    
+    @Dependency(StudyManager.self)
+    private var studyManager
+    
+    @Dependency(BulkHealthExporter.self)
+    private var bulkExporter
+    
+    @Application(\.logger)
+    private var logger
+    
+    private(set) var session: (any BulkExportSession<HistoricalSamplesToFHIRJSONProcessor>)?
+    
+    
+    func configure() {
+        startAutomaticExportingIfNeeded()
+        scheduleOrphanedExportsForUpload()
+    }
+    
+    
+    /// Schedules all files in the bulkExports folder to be uploaded, unless they have already been scheduled.
+    ///
+    /// The purpose of this function is to allow us to retry any unsucessful uploads, which failed bc the app was quit, or for some other reason.
+    private func scheduleOrphanedExportsForUpload() {
+        let urls = (try? FileManager.default.contentsOfDirectory(at: .scheduledHealthKitUploads, includingPropertiesForKeys: nil)) ?? []
+        guard !urls.isEmpty else {
+            return
+        }
+        Task.detached(priority: .utility) {
+            await withDiscardingTaskGroup { taskGroup in
+                for url in urls {
+                    taskGroup.addTask(priority: .utility) {
+                        try? await self.uploadAndDelete(url)
+                    }
+                }
+            }
+        }
+    }
+    
+    
+    /// Starts the automatic collection of historical health data,
+    /// unless it's already running, or automatic collection is disabled via ``FeatureFlags/disableAutomaticBulkHealthExport``.
+    nonisolated func startAutomaticExportingIfNeeded() {
+        Task {
+            await setupAndStartExportSession()
+        }
+    }
+    
+    
+    /// Cancels the session, deletes all progress associated with it, and restarts it
+    ///
+    /// - Note: This is intended primarily for debugging purposes
+    func fullyResetSession() throws {
+        try bulkExporter.deleteSessionRestorationInfo(for: .mhcHistoricalDataExport)
+        Task { @Sendable in
+            while let session = self.session, session.state != .terminated {
+                try await Task.sleep(for: .seconds(1))
+            }
+            self.session = nil
+            await self.setupAndStartExportSession()
+        }
+    }
+    
+    
+    /// - returns: A Boolean indicating whether the session was successfully set up and started
+    @discardableResult
+    private func setupAndStartExportSession() async -> Bool {
+        if session == nil {
+            guard let study = studyManager.studyEnrollments.first?.study else {
+                logger.error("\(#function) aborting: no study")
+                return false
+            }
+            do {
+                session = try await bulkExporter.session(
+                    withId: .mhcHistoricalDataExport,
+                    for: study.allCollectedHealthData,
+                    startDate: .oldestSample,
```

Comment:
I think we aligned on the last ten years for each users.

Reply:
anders mentioned 5 years in the doc, but also said that he wasn't sure about it; i'll change it to 5 for the time being and open an issue to track this. we can easily adjust it once we know the exact time range